ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ntp.j2

```diff
@@ -1,16 +1,61 @@
 {# eos - NTP #}
-{% if ntp.authenticate is arista.avd.defined(true) %}
+{% if ntp is arista.avd.defined %}
+!
+{%     for local_interface in ntp.local_interfaces | arista.avd.natural_sort('name') %}
+{%         if local_interface.name is arista.avd.defined %}
+{%             set ntp_int_cli = "ntp local-interface" %}
+{%             if local_interface.vrf is arista.avd.defined and local_interface.vrf != 'default' %}
+{%                 set ntp_int_cli = ntp_int_cli ~ " vrf " ~ local_interface.vrf %}
+{%             endif %}
+{%             set ntp_int_cli = ntp_int_cli ~ " " ~ local_interface.name %}
+{{ ntp_int_cli }}
+{%         endif %}
+{%     endfor %}
+{%     for server in ntp.servers | arista.avd.natural_sort('name') %}
+{%         if server.name is arista.avd.defined %}
+{%             set ntp_server_cli = "ntp server" %}
+{%             if server.vrf is arista.avd.defined and server.vrf != 'default' %}
+{%                 set ntp_server_cli = ntp_server_cli ~ " vrf " ~ server.vrf %}
+{%             endif %}
+{%             set ntp_server_cli = ntp_server_cli ~ " " ~ server.name %}
+{%             if server.preferred is arista.avd.defined(true) %}
+{%                 set ntp_server_cli = ntp_server_cli ~ " prefer" %}
+{%             endif %}
+{%             if server.burst is arista.avd.defined(true) %}
+{%                 set ntp_server_cli = ntp_server_cli ~ " burst" %}
+{%             endif %}
+{%             if server.iburst is arista.avd.defined(true) %}
+{%                 set ntp_server_cli = ntp_server_cli ~ " iburst" %}
+{%             endif %}
+{%             if server.version is arista.avd.defined %}
+{%                 set ntp_server_cli = ntp_server_cli ~ " version " ~ server.version %}
+{%             endif %}
+{%             if server.minpoll is arista.avd.defined %}
+{%                 set ntp_server_cli = ntp_server_cli ~ " minpoll " ~ server.minpoll %}
+{%             endif %}
+{%             if server.maxpoll is arista.avd.defined %}
+{%                 set ntp_server_cli = ntp_server_cli ~ " maxpoll " ~ server.maxpoll %}
+{%             endif %}
+{%             if server.local_interface is arista.avd.defined %}
+{%                 set ntp_server_cli = ntp_server_cli ~ " local-interface " ~ server.local_interface %}
+{%             endif %}
+{%             if server.key is arista.avd.defined %}
+{%                 set ntp_server_cli = ntp_server_cli ~ " key " ~ server.key %}
+{%             endif %}
+{{ ntp_server_cli }}
+{%         endif %}
+{%     endfor %}
+{%     for authentication_key in ntp.authentication_keys | arista.avd.natural_sort('id') %}
```

Comment:
All NTP configuration should be rendered before NTP servers, as seen in EOS:
```
POD1_SPINE01(config)#ntp local-interface lo1
POD1_SPINE01(config)#ntp server 1.2.3.4 local-interface lo0
POD1_SPINE01(config)#ntp server 2.2.2.55
POD1_SPINE01(config)#ntp server 10.1.1.1
POD1_SPINE01(config)#ntp server 10.1.1.2 prefer
POD1_SPINE01(config)#ntp server 20.20.20.1 key 2
POD1_SPINE01(config)#ntp server ie.pool.ntp.org iburst key 1
POD1_SPINE01(config)#ntp authentication-key 1 md5 044F0E151B
POD1_SPINE01(config)#ntp authentication-key 2 sha1 15060E1F10
POD1_SPINE01(config)#ntp trusted-key 1-2
POD1_SPINE01(config)#ntp authenticate
POD1_SPINE01(config)#sh run | i ntp
ntp authentication-key 2 sha1 7 075E741C18593C5431435B
ntp authentication-key 1 md5 7 040B5F522971691F5C4827
ntp trusted-key 1-2
ntp authenticate
ntp local-interface Loopback1
ntp server 1.2.3.4 local-interface Loopback0
ntp server 2.2.2.55
ntp server 10.1.1.1
ntp server 10.1.1.2 prefer
ntp server 20.20.20.1 key 2
ntp server ie.pool.ntp.org iburst key 1
```

Reply:
Ack. Fix on it's way.